### PR TITLE
[FIX] spreadsheet: implement ODOO.FILTER.VALUE for from/to date filter

### PR DESCRIPTION
**Pull request by LucasLefevre**
Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/spreadsheet/static/src/global_filters/plugins/global_filters_ui_plugin.js [hunk 1]
```diff
@@ -229,17 +230,29 @@ export class GlobalFiltersUIPlugin extends spreadsheet.UIPlugin {
         const value = this.getGlobalFilterValue(filter.id);
         switch (filter.type) {
             case "text":
-                return value || "";
+                return { value: value || "" };
```
Comment (hokolomopo) [inline addons/spreadsheet/static/src/global_filters/plugins/global_filters_ui_plugin.js:233]: maybe return an array every time ? woulad avoid one isArray()

Comment (hokolomopo) [reply addons/spreadsheet/static/src/global_filters/plugins/global_filters_ui_plugin.js:233]: also the fucntion doesn't return a displayValue anymore, we should rename it

File: addons/spreadsheet/static/src/global_filters/plugins/global_filters_ui_plugin.js [hunk 2]
```diff
@@ -532,20 +545,40 @@ export class GlobalFiltersUIPlugin extends spreadsheet.UIPlugin {
...
+        let filterRowIndex = 1; // first row is the column titles
+        const locale = this.getters.getLocale();
         for (const filter of this.getters.getGlobalFilters()) {
-            const content = this.getters.getFilterDisplayValue(filter.label);
-            cells[`A${row}`] = { content: filter.label };
-            cells[`B${row}`] = { content };
-            row++;
+            const result = this.getFilterDisplayValue(filter.label);
+            cells[`A${filterRowIndex + 1}`] = { content: filter.label };
+            // TODO use isMatrix
```
Comment (hokolomopo) [inline addons/spreadsheet/static/src/global_filters/plugins/global_filters_ui_plugin.js:556]: todo

File: addons/spreadsheet/static/src/global_filters/plugins/global_filters_ui_plugin.js (hunk 1 above)
Comment (rrahir) [reply addons/spreadsheet/static/src/global_filters/plugins/global_filters_ui_plugin.js:233]: > maybe return an array every time ? woulad avoid one isArray()

+1

**Review by pro-odoo — APPROVED** [review]
robodoo rebase-ff r+

**Comment by pro-odoo** [comment]
robodoo r- (conflict)

**Pull request closed without merging**